cli/spinner.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+const encoder = new TextEncoder();
+
+const LINE_CLEAR = encoder.encode("\r\u001b[K"); // From cli/prompt_secret.ts
+const COLOR_RESET = "\u001b[0m";
+const DEFAULT_SPEED = 75;
+
+/** The options for the spinner */
+export interface SpinnerOptions {
+  /** The spinner to be animated */
+  spinner: string[];
+  /** The message to display next to the spinner */
+  message?: string;
+  /** The speed of the spinner. Defaults to 75 */
+  speed?: number;
+  /** The color of the spinner. Defaults to the default terminal color */
+  color?: string;
+  /** Whether to cleanup the output if the process is terminated. Defaults to true */
+  cleanup?: boolean;
+}
+
+/** The default spinners */
+export const spinners = {
+  lines: ['-', '\\', '|', '/'],
+  binary: ['0000', '1000', '1100', '1110', '1111', '0111', '0011', '0001'],
+  dots: ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏'],
+  dots2: ['⣾', '⣽', '⣻', '⢿', '⡿', '⣟', '⣯', '⣷'],
+  blink: ['◜', '◠', '◝', '◞', '◡', '◟'],
+}
+
+/**
+ * A spinner that can be used to indicate that something is loading.
+ */
+export class Spinner {
+  #message: string | undefined;
+  #speed: number;
+  #spinner: string[];
+  #color: string | undefined;
+  #cleanup: boolean;
+  
+  #intervalId: number | undefined;
+
+  /**
+   * Creates a new spinner.
+   * @param message - The message to display next to the spinner.
+   * @param speed - The speed of the spinner.
+   * @param spinner - The spinner to use.
+   * @param color - The color of the spinner.
+   * @param cleanup - Whether to cleanup the output if the process is terminated.
+   */
```

Comment:
These JSDoc directives aren't needed if we go with the below suggestion. Are you able to provide an `@example` block here?